wasatch/SPIDevice.py

```diff
@@ -0,0 +1,331 @@
+            binned = [(value + next_value)/2 for value, next_value in zip(reading.spectrum[:-1], next_idx_values)]
+            binned.append(reading.spectrum[-1])
+            reading.spectrum = binned
+                
+
+        self.session_reading_count += 1
+        reading.session_count = self.session_reading_count
+        reading.sum_count = self.sum_count
+
+        if averaging_enabled:
+            if self.sum_count >= self.settings.state.scans_to_average:
+                reading.spectrum = [ x / self.sum_count for x in self.summed_spectra ]
+                log.debug("spi device acquire data: averaged_spectrum : %s ...", reading.spectrum[0:9])
+                reading.averaged = True
+
+                # reset for next average
+                self.summed_spectra = None
+                self.sum_count = 0
+        else:
+            # if averaging isn't enabled...then a single reading is the
+            # "averaged" final measurement (check reading.sum_count to confirm)
+            reading.averaged = True
+
+        return SpectrometerResponse(reading)
+
+    def write_eeprom(self) -> SpectrometerResponse:
+        try:
+            self.settings.eeprom.generate_write_buffers()
+        except:
+            log.critical("failed to render EEPROM write buffers", exc_info=1)
+            #self.message_queue("marquee_error", "Failed to write EEPROM")
+            return False
+
+        for page in range(EEPROM.MAX_PAGES):
+            self._EEPROMWritePage(page,self.settings.eeprom.write_buffers[page])
+
+        #self.message_queue("marquee_info", "EEPROM successfully updated")
+        return SpectrometerResponse(True)
+
+    def _EEPROMReadPage(self, page: int) -> list[bytes]:
+        while True:
+            EEPROMPage  = bytearray(74)
+            command     = bytearray(7)
+            command     = [0x3C, 0x00, 0x02, 0xB0, (0x40 + page), 0xFF, 0x3E]
+            self.SPI.write(command, 0, 7)
+            time.sleep(0.01)
+            command = [0x3C, 0x00, 0x01, 0x31, 0xFF, 0x3E]
+            self.SPI.write_readinto(command, EEPROMPage)
+            log.debug(f"read eeprom page of {EEPROMPage}")
+            if EEPROMPage[:6] == [0x3,0x3,0x3,0x3,0x3,0x3]:
+                log.info(f"trying to read page {page}, got response status of {EEPROMPage[:6]}")
+                continue
+            else:
+                break
+        return EEPROMPage
+
+    def set_integration_time_ms(self, value: int) -> SpectrometerResponse:
+        while self.acquiring:
+            time.sleep(0.01)
+            continue
+        self._SPIWrite(value, self.INTEGRATION_ADDRESS)
+        self.settings.state.integration_time_ms = value
+        return SpectrometerResponse()
+
+    def _SPIWrite(self, value: int, address: int) -> None:
+        command = bytearray(8)
+        # Convert the int into bytes.
+        txData = bytearray(2)
+        txData[1]   = value >> 8
+        txData[0]   = value - (txData[1] << 8)
```

Comment:
It _looks_ like you're assuming int means uint16?  That would be okay for most cases.